cli/src/commands/git.rs

```diff
@@ -1049,7 +1049,12 @@ fn cmd_git_push(
         branch_updates,
         force_pushed_branches,
     };
-    with_remote_git_callbacks(ui, |cb| {
+    let mut scratch: Vec<u8> = Vec::new();
```

Comment:
nit: `scratch` buffer could be moved to the sideband handler if the caller doesn't flush the remainder at the end of the call.

```rust
    let mut scratch: Vec<u8> = Vec::new();
    move |ui: &Ui, progress_message: &[u8]| {
```

Alternatively, `get_git_sideband_progress_message_writer()` could be a struct having `.write(ui, progress_message)` and `.flush()` methods.

Reply:
Thanks for the reminder, I realized I forgot to flush the remainder of the buffer. Updated this to use a struct.